### PR TITLE
PF-435: Confirmation prompt for `delete` commands.

### PR DESCRIPTION
**Pull request by marikomedlock**
- Added a confirmation prompt for all `delete` commands: `groups delete`, `resources delete`, `workspace delete`.
  ```Are you sure you want to delete (y/N)?```
- Added an optional `--quiet` flag to suppress the interactive prompt.
- Updated all existing tests to use the `--quiet` flag. Added new tests for the interactive option.

### CODE REVIEW

File: src/main/java/bio/terra/cli/command/shared/options/DeletePrompt.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+public class DeletePrompt {
+
+  @CommandLine.Option(names = "--quiet", description = "Suppress interactive prompt and delete.")
+  private boolean quiet;
+
+  /**
+   * Helper method to generate an interactive confirmation prompt if the `--quiet` option is not
+   * specified. Throws a {@link UserActionableException} if the prompt response is negative.
+   */
+  public void throwIfConfirmationPromptNegative() {
```
Comment (jaycarlton) [inline src/main/java/bio/terra/cli/command/shared/options/DeletePrompt.java:28]: maybe just `confirmOrThrow`

File: src/main/java/bio/terra/cli/utils/Printer.java [hunk 2]
```diff
@@ -20,16 +21,19 @@
...
 
   private final PrintStream out;
   private final PrintStream err;
+  private final InputStream in;
 
   private static Printer printer;
 
   /** Constructor that initializes the printer with the specified output streams. */
-  private Printer(PrintStream out, PrintStream err) {
+  private Printer(PrintStream out, PrintStream err, InputStream in) {
```
Comment (jaycarlton) [inline src/main/java/bio/terra/cli/utils/Printer.java:33]: Maybe need to rename the class now that it's reading in addition to printing.

File: src/main/java/bio/terra/cli/command/shared/options/DeletePrompt.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    boolean promptReturnedYes = true; // default to Y=yes if user input is suppressed
+    PrintStream OUT = Printer.getOut();
+    if (!quiet) {
+      OUT.print("Are you sure you want to delete (y/N)? ");
```
Comment (jaycarlton) [inline src/main/java/bio/terra/cli/command/shared/options/DeletePrompt.java:32]: scope creep: It might be nice to display some metadata about the thing to be deleted, such as how many resources are contained within a workspace, objects in a bucket, etc.

**Review by jaycarlton — APPROVED** [review]

**Review by jmczerk — APPROVED** [review]
LGTM

File: src/main/java/bio/terra/cli/command/shared/options/DeletePrompt.java (hunk 1 above)
Comment (marikomedlock) [reply src/main/java/bio/terra/cli/command/shared/options/DeletePrompt.java:28]: Changed.

File: src/main/java/bio/terra/cli/utils/Printer.java (hunk 2 above)
Comment (marikomedlock) [reply src/main/java/bio/terra/cli/utils/Printer.java:33]: Renamed to `UserIO`.

File: src/main/java/bio/terra/cli/command/shared/options/DeletePrompt.java (hunk 3 above)
Comment (marikomedlock) [reply src/main/java/bio/terra/cli/command/shared/options/DeletePrompt.java:32]: It's a good suggestion, but I think out of scope for this PR/ticket which is just to prompt for confirmation before deleting. Filed [PF-891](https://broadworkbench.atlassian.net/browse/PF-891).

**Comment by marikomedlock** [comment]
There was 1 unit test failure with a SAM socket timeout exception: ClientExceptionHandling. I ran it locally and it passed.
